lib/package

```diff
@@ -0,0 +1,55 @@
+function fetch_engine_package() {
+    local engine="$1"
+    local version="$2"
+    local location="$3"
+
+    local base_url="$PHP_BASE_URL"
+    if [ "$engine" = "nginx" ] ; then
+      base_url="$NGINX_BASE_URL"
+    fi
+
+    fetch_package "$base_url" "${engine}-${version}" "$location"
+}
+
+function has_package() {
```

Comment:
best practice: I'm not sure about what this function does. Maybe add a comment or rename it so that it's clearer? If I understand correctly it makes a HEAD request to our object storage to check if the given package exists, which means that we pre-compiled the extension, isn't it?